### ISSUE TITLE
Reproducing results

### ISSUE THREAD

**Issue by praxidike** [tdeboissiere/DeepLearningImplementations#41]
I am trying to reproduce the results from the pix2pix model and the facades dataset. However the results always look kind of messed up. The following image was generated after more than 300 epochs of training. These monochrome areas keep on appearing in every image.

![grafik](https://user-images.githubusercontent.com/14093884/28305289-62e8faf6-6b9b-11e7-87e6-f088c7347538.png)

Any hints what might have gone wrong? I use the used the exact same code here from GitHub.

**Comment by tdeboissiere** [comment]
Hard to say, but 2 hints:

- It may be that when looking calling the script to create the generated images, you only call it on one image, and since the Batch Norm mode is configured to run as in training, the batch norm statistics are heavily skewed when computed on a single sample
- Try overfitting on a very small dataset with a very large weight on the per pixel loss. If you don't get good results, this might give you ideas.

**Comment by jiangzidong** [comment]
Maybe its your image-draw functions' issue, because I can reproduce the good result (backend tensorflow, patch size 64)
what about the picture in pix2pix/figures/current_batch_training.png  or current_batch_validation.png

**Comment by AlexanderFabisch** [comment]
I can reproduce this problem. [Here](https://gist.github.com/AlexanderFabisch/da3a5b5cda73bbfecfb6bf1390248c5b) is the code that I used to generate images. This is the result that I got:

![generated](https://user-images.githubusercontent.com/869592/28914464-0ad04074-783c-11e7-9ab0-9165be431263.png)

The pix2pix/figures/current_batch_validation.png however looks like this:

![validation](https://user-images.githubusercontent.com/869592/28914504-30578e88-783c-11e7-9a84-38e8a937fa02.png)

How is this generated? I cannot find it in the code.

**Comment by AlexanderFabisch** [comment]
OK, I found it in the code: pix2pix/src/utils/data_utils.py

I added inverse normalization and now I got this:

![generated](https://user-images.githubusercontent.com/869592/28915404-52517564-783f-11e7-8633-54382b2b28f3.png)

Is there anything else that I did not see? I use a batch size of 4 and a patch size of 64x64.

**Comment by tdeboissiere** [comment]
What happens if you increase the batch size ?
When you said you applied inverse normalization, did you check that our images where uint8, pixel amplitude bounded between [0, 255] ?

**Comment by AlexanderFabisch** [comment]
It was indeed a problem with the inverse normalization (not the batch normalization). I fixed it and now the results look really great after 150 epochs on the validation set. Thanks for your help!

![generated](https://user-images.githubusercontent.com/869592/28958975-cf0a503a-78f8-11e7-97d6-5d3360b26f01.png)
